padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,216 @@
+        self.n_layer = 3
+        self.padding = 1
+        self.size_bottleneck = 28  # image size 28*28
+        stride = int(kernel_size/2)
+
+        for idx in range(self.n_layer):
+            self.size_bottleneck = int(
+                (self.size_bottleneck + 2 * self.padding - (kernel_size - 1) - 1) / stride + 1)
+
+        # encoder
+        self.conv_encoder = nn.Sequential(
+            torch.nn.Conv2d(in_channels=1, out_channels=hidden_channels, kernel_size=kernel_size, stride=stride,
+                            padding=self.padding),
+            activation_function(),
+            torch.nn.Conv2d(in_channels=hidden_channels, out_channels=hidden_channels, kernel_size=kernel_size,
+                            stride=stride, padding=self.padding),
+            activation_function(),
+            torch.nn.Conv2d(in_channels=hidden_channels, out_channels=1, kernel_size=kernel_size, stride=stride,
+                            padding=self.padding),
+            activation_function(),
+        )
+        self.linear_1 = torch.nn.Linear(in_features=self.size_bottleneck**2, out_features=1)
+        self.linear_2 = torch.nn.Linear(in_features=self.size_bottleneck**2, out_features=1)
+
+        # decoder
+        self.transp_conv_decoder = nn.Sequential(
+            torch.nn.ConvTranspose2d(in_channels=1, out_channels=hidden_channels, kernel_size=kernel_size,
+                                     stride=stride),
+            activation_function(),
+            torch.nn.ConvTranspose2d(in_channels=hidden_channels, out_channels=hidden_channels,
+                                     kernel_size=kernel_size, stride=stride),
+            activation_function(),
+            torch.nn.ConvTranspose2d(in_channels=hidden_channels, out_channels=1, kernel_size=kernel_size,
+                                     stride=stride),
+        )
+        self.linear_3 = torch.nn.Linear(in_features=1, out_features=self.size_bottleneck**2)
+
+    def encode(self, x):
+        x = self.conv_encoder(x)
+        x = rearrange(x, 'b c x y -> b (c x y)')
+        mu = self.linear_1(x)  # mean value
+        std = 0.5 * self.linear_2(x)  # standard deviation
+        eps = torch.randn_like(std)
+        return mu + eps*std, mu, std
+
+    def decode(self, x):
+        x = self.linear_3(x)
+        x = rearrange(x, 'b (x y) -> b x y', y=self.size_bottleneck, x=self.size_bottleneck)
+        x = x[:, None, :, :]
+        return self.transp_conv_decoder(x)
+
+    def forward(self, input):
+        x, mu, std = self.encode(input)
+        y = self.decode(x)
+        return y[:, :, :28, :28], mu, std
+
+
+@ex.config
+def config():
+    epochs = 2          # Use resume=True to train for more epochs
+    batch_size = 10
+    resume = False      # PT: Continue from checkpoints
+
+    assert os.environ.get('STORAGE_ROOT') is not None, 'STORAGE_ROOT is not defined in the environment variables'
+    logfilepath = os.environ['STORAGE_ROOT'] + '/vae_mnist'  # PT creates log files
```

Comment:
I would personally prefer that each new experiment uses a new subdirectory in `os.environ['STORAGE_ROOT'] + '/vae_mnist'`

Reply:
This should be adapted to our new storage root behavior as showcased in the other examples.